### ISSUE TITLE
Reapit Feed not Showing Properties belonging to Agent

### ISSUE THREAD

**Issue by mattpbs** [reapit/foundations#5735]
Not sure if this is a bug but I have been asked to report here.

I have gotten my client (DB Roberts) to connect his Reapit Agent account to our system using Reapit Connect.

When pulling a list of properties using https://platform.reapit.cloud/properties endpoint we get a whole list of properties. However, none of these are from DB Roberts' account.

I'm assuming these are example properties for development purposes but this is supposed to now be in a production environment so I expecting to see their actual properties.

Is there an issue or is there something we have missed?

Thanks

Matt

**Comment by plittlewood-rpt** [comment]
Hi @mattpbs - can you confirm if DB Robert’s have installed your app? If not it’s likely you are seeing sandbox data. I’d need to understand more about your app to be able to give a proper steer. Is your app machine to machine (client credentials auth) or does it have a user in context (authorisation code auth). Is your app listed? If not you’ll definitely only be seeing sandbox data as until it is listed customers cannot install the app anyway.

**Comment by mattpbs** [comment]
Hi @plittlewood-rpt 

Unfortunately, I don't have access to DB Roberts's account to check but I believe so as they can see the App (This is a private app). They have been dealing with Shahzahan from your account team who has been assisting them and I believe this is all set up.

There is nothing too complicated about the app. We are using Reapit connect to control the Oauth authentication and then all we need to do is pull a list of properties from their system.

**Comment by plittlewood-rpt** [comment]
Hi @mattpbs - can you confirm if the properties you are seeing are indeed those from the sandbox? Are you using client credentials or authorization code OAuth grant when authenticating?

**Comment by plittlewood-rpt** [comment]
Hi @mattpbs - can you confirm if this issue is still outstanding, and if so come back to me on the above?

**Comment by mattpbs** [comment]
Hi @plittlewood-rpt 

Yes, issue is still outstanding. The properties we are seeing are indeed from sandbox. We are using client credentials to authorise.

**Comment by plittlewood-rpt** [comment]
Hi @mattpbs thanks - if you are using client credentials grant to authorize you need to pass the correct `reapit-customer` header with your request. You'll currently be passing SBOX. You can see the ids of the customers that have installed your app on the app page in the Developer portal. See https://foundations-documentation.reapit.cloud/api/api-documentation#accessing-customer-data 

Let me know if you need any further help

**Comment by mattpbs** [comment]
Hi @plittlewood-rpt 

We are already sending the header - reapit-customer: DBR. however in the admin it says there are no installations of your app so guessing they haven't installed it correctly?

**Comment by plittlewood-rpt** [comment]
Hi @mattpbs - can you confirm the id or name of your app so I can have a look please?

**Comment by mattpbs** [comment]
@plittlewood-rpt The id in url is 430c1657-cf11-4467-9c75-e59ed70d6490 and client id is 6ollgs789v2simvtpcci6isano

**Comment by plittlewood-rpt** [comment]
Hi @mattpbs thanks - your app is configured to use authorization_code grant so the `reapit-customer` header won't do anything which is why you are seeing Sandbox data. It will need to be reregistered using the correct grant type if you are wanting it to be a serverside application. The app is also not currently listed so won't be available for the customer to install. Please see the documentation at https://foundations-documentation.reapit.cloud/listing-your-app which runs through what you need to do to get your app listed. Hope this helps

**Comment by mattpbs** [comment]
Thanks @plittlewood-rpt

We have a client marketplace for integrations so it makes sense to reuse what we had and use Authorization Code flow and use Reapit Connect to authorise in this way. We can use Client Credentials flow if necessary but I am hoping to avoid this as it'll require extra work.

The app has been submitted for approval but doesn't say anything different on the status. According to the client it showed a message saying about entering billing settings so wondering if this needs to be done first.

**Comment by plittlewood-rpt** [comment]
Hi @mattpbs - it doesn't look like this app has been submitted for approval yet. If you have a look through the steps at the link I sent earlier, you should be able to get it submitted for out team to review

**Comment by mattpbs** [comment]
Hi @plittlewood-rpt. As far as we're aware the app has been submitted for approval. Do we need to make sure the client fills out the billing section before the system will actually submit it?

**Comment by plittlewood-rpt** [comment]
Hi @mattpbs - the billing section does need to be completed yes before you can submit the app. When you submit your app you need to set the Listed flag to checked/on which is the thing that will make it available to customers.

**Comment by plittlewood-rpt** [comment]
Hi @mattpbs - as there has been no movement on this ticket we will close it. Please review the comments above which will allow you to complete your app for submission.

**Comment by mattpbs** [comment]
Hi @plittlewood-rpt 

We've been through now and the app is now finally approved and we can see that it is installed by DB Roberts so that bit looks like it's all working.

We have gotten the client to reauthorize the account using Authentication code flow, but the data that is showing still appears to be sandbox data. I understand from the documentation and from what you have said that the reapit-customer header only applies to Client credentials flow so wouldn't work in our case.

Do you have any other ideas?

Thanks

Matt
